backend/src/main/java/com/bakdata/conquery/models/query/preview/EntityPreviewExecution.java

```diff
@@ -132,72 +136,74 @@ private List<EntityPreviewStatus.Info> transformQueryResultToInfos(ManagedQuery
-		final List<EntityPreviewStatus.TimebasedInfos> timebasedInfos = new ArrayList<>();
+	private List<EntityPreviewStatus.ChronoInfos> toChronoInfos(PreviewConfig previewConfig, Map<String, ManagedQuery> subQueries, PrintSettings printSettings) {
+		final List<EntityPreviewStatus.ChronoInfos> chronoInfos = new ArrayList<>();
+
+		for (PreviewConfig.ChronoSelects description : previewConfig.getChronoSelects()) {
+			final ManagedQuery query = subQueries.get(description.name());
+
 
-		for (PreviewConfig.TimebasedSelects description : previewConfig.getTimebasedSelects()) {
-			final ManagedQuery query = subQueries.get(description.name()).get(0);
 			final EntityResult entityResult = query.streamResults().collect(MoreCollectors.onlyElement());
 
 			final Map<SelectId, PreviewConfig.InfoCardSelect> select2desc =
-					description.selects().stream()
-							   .collect(Collectors.toMap(PreviewConfig.InfoCardSelect::select, Function.identity()));
+					description.selects().stream().collect(Collectors.toMap(PreviewConfig.InfoCardSelect::select, Function.identity()));
 
-			//TODO make constants of AbsoluteExportForm
-			final int resolutionInfoIdx = 0;
-			final int timeIdx = 2;
 
 			// Group lines by quarter, year and resolution.
 			final Map<Integer, Object[]> yearLines = new HashMap<>();
 			final Map<Integer, Map<Integer, Object[]>> quarterLines = new HashMap<>();
 
-			groupLinesForResolutions(entityResult, resolutionInfoIdx, timeIdx, yearLines, quarterLines);
+			groupLinesForResolutions(entityResult, yearLines, quarterLines);
 
 			final Function<Object[], Map<String, Object>> lineTransformer = createLineTransformer(query.getResultInfos(), select2desc, printSettings);
 
-			final List<EntityPreviewStatus.TimebasedInfos.YearEntry> yearEntries = new ArrayList<>();
+			final List<EntityPreviewStatus.YearEntry> yearEntries = new ArrayList<>();
 
 			yearLines.forEach((year, yearLine) -> {
 
-				final List<EntityPreviewStatus.TimebasedInfos.QuarterEntry> quarterEntries = new ArrayList<>();
+				final List<EntityPreviewStatus.QuarterEntry> quarterEntries = new ArrayList<>();
 				quarterLines.getOrDefault(year, Collections.emptyMap())
-							.forEach((quarter, line) -> quarterEntries.add(new EntityPreviewStatus.TimebasedInfos.QuarterEntry(quarter, lineTransformer.apply(line))));
+							.forEach((quarter, line) -> quarterEntries.add(new EntityPreviewStatus.QuarterEntry(quarter, lineTransformer.apply(line))));
 
-				quarterEntries.sort(Comparator.comparingInt(EntityPreviewStatus.TimebasedInfos.QuarterEntry::quarter));
+				quarterEntries.sort(Comparator.comparingInt(EntityPreviewStatus.QuarterEntry::quarter));
 
 
-				yearEntries.add(new EntityPreviewStatus.TimebasedInfos.YearEntry(year, lineTransformer.apply(yearLine), quarterEntries));
+				yearEntries.add(new EntityPreviewStatus.YearEntry(year, lineTransformer.apply(yearLine), quarterEntries));
 			});
```

Comment:
Bitte Doku, weshalb hier quarter und year gegen einander verarbeitet werden müssen.

Reply:
da gibt es keinen technischen Grund, das war einfach die abgemachte API?

Reply:
Oki, was ich mich gefragt habe war warum das nicht in `groupLinesForResolutions` ist. An sich nimmt dieser Code-Block doch `yearLines` & `quarterLines` und gibt `yearEntries` aus. Kannst du das in eine Funktion packen?

Reply:
ah, ja guter Punkt